app/scripts/modules/kubernetes/src/container/environmentFrom.component.ts

```diff
@@ -0,0 +1,80 @@
+import { IComponentOptions, IController, module } from 'angular';
+
+interface IKubernetesEnvironmentFrom {
+  prefix: string;
+  configMapRef?: IKubernetesConfigMapRef;
+  secretRef?: IKubernetesSecretRef;
+}
+
+interface IKubernetesConfigMapRef {
+  name: string;
+  optional: boolean;
+}
+
+interface IKubernetesSecretRef {
+  name: string;
+  optional: boolean;
+}
+
+const CONFIGMAP_TYPE = 'Config Map';
+const SECRET_TYPE = 'Secret';
+const DEFAULT_ENVFROM_TYPE = CONFIGMAP_TYPE;
+
+class KubernetesEnvironmentFromCtrl implements IController {
+  public envFrom: IKubernetesEnvironmentFrom[];
+  public sourceTypes: string[];
+  public envFromSourceTypes: string[];
+
+  public $onInit(): void {
+    if (!this.envFrom ) {
+      this.envFrom = [];
+    }
+    this.sourceTypes = [CONFIGMAP_TYPE, SECRET_TYPE];
+    this.mapEnvFromSourceTypes();
+  }
+
+  public mapEnvFromSourceTypes(): void {
+    this.envFromSourceTypes = this.envFrom
+      .map((envFrom) => {
+        if (envFrom.configMapRef) {
+          return CONFIGMAP_TYPE;
+        } else {
+          return SECRET_TYPE;
+        }
+      });
+  }
+
+  public removeEnvFrom(index: number): void {
+    this.envFrom.splice(index, 1);
+    this.envFromSourceTypes.splice(index, 1);
+  }
+
+  public addEnvFrom(index: number): void {
+    this.envFrom.push({});
```

Comment:
What you push here has to be of type `IKubernetesEnvironmentFrom`: `this.envFrom.push({ prefix: '' })`.